File: scripts/patch-project-view-keywords.mjs
#!/usr/bin/env node
/**
 * Restore the semiconductor view and keep the vein-closure view cost-project-only.
 * Usage: node scripts/patch-project-view-keywords.mjs /path/to/daily-report-digest.config.json
 */
import fs from "node:fs";

const VIEW_PATCH = {
  cla: { label: "CLA", keyword: "CLA" },
  oct: { label: "OCT", keyword: "OCT" },
  "laser-shockwave": { label: "冲击波", keyword: "冲击波" },
  "large-vessel-plaque": { label: "斑块减容", keyword: "斑块减容" },
  "semiconductor-vein": { label: "半导体", keyword: "半导体" },
  "vein-closure-system": { label: "静脉腔闭合系统", costProjectContains: "静脉腔" },
};

const path = process.argv[2] || "data/daily-report-digest.config.json";
const cfg = JSON.parse(fs.readFileSync(path, "utf8"));
for (const org of cfg.orgs ?? []) {
  for (const pv of org.projectViews ?? []) {
    const patch = VIEW_PATCH[pv.id];
    if (!patch) continue;
    pv.label = patch.label;
    pv.filters = pv.filters || {};
    if (patch.keyword) {
      pv.filters.keyword = patch.keyword;
      delete pv.filters.workModuleContains;
      delete pv.filters.costProjectContains;
    }
    else {
      delete pv.filters.keyword;
      delete pv.filters.workModuleContains;
      pv.filters.costProjectContains = patch.costProjectContains;
    }
  }
}
fs.writeFileSync(path, JSON.stringify(cfg, null, 2) + "\n");
console.log("patched label+keyword for", Object.keys(VIEW_PATCH).join(", "));

File: scripts/patch-project-view-keywords.py
#!/usr/bin/env python3
"""Restore the semiconductor view and keep the vein-closure view cost-project-only."""
import json
import sys
from pathlib import Path

VIEW_PATCH = {
    "cla": {"label": "CLA", "keyword": "CLA"},
    "oct": {"label": "OCT", "keyword": "OCT"},
    "laser-shockwave": {"label": "冲击波", "keyword": "冲击波"},
    "large-vessel-plaque": {"label": "斑块减容", "keyword": "斑块减容"},
    "semiconductor-vein": {"label": "半导体", "keyword": "半导体"},
    "vein-closure-system": {"label": "静脉腔闭合系统", "costProjectContains": "静脉腔"},
}

path = Path(sys.argv[1] if len(sys.argv) > 1 else "data/daily-report-digest.config.json")
cfg = json.loads(path.read_text(encoding="utf-8"))
for org in cfg.get("orgs") or []:
    for pv in org.get("projectViews") or []:
        patch = VIEW_PATCH.get(pv.get("id"))
        if not patch:
            continue
        pv["label"] = patch["label"]
        filters = pv.setdefault("filters", {})
        if "keyword" in patch:
            filters["keyword"] = patch["keyword"]
            filters.pop("workModuleContains", None)
            filters.pop("costProjectContains", None)
        else:
            filters.pop("keyword", None)
            filters.pop("workModuleContains", None)
            filters["costProjectContains"] = patch["costProjectContains"]
path.write_text(json.dumps(cfg, ensure_ascii=False, indent=2) + "\n", encoding="utf-8")
print("patched label+keyword for", ", ".join(VIEW_PATCH.keys()))
